fud2/fud-core/src/exec/driver.rs

```diff
@@ -203,7 +204,9 @@ impl Driver {
     /// Print a list of registered states and operations to stdout.
     pub fn print_info(&self) {
         println!("States:");
-        for (_, state) in self.states.iter() {
+        for (_, state) in
+            self.states.iter().sorted_by_key(|(_, state)| &state.name)
+        {
```

Comment:
It may very well be a good idea to sort stuff alphabetically, but I just wanted to point out that there is actually a signal embedded in the original order of these things (both states and ops). When searching for plans, fud2 breaks ties based on ordering. So, for example, it's possible to register multiple states with the extension `.sv` (different forms of Verilog emitted from different compilation strategies, for instance). But when guessing the state for a file `foo.sv`, the first state wins.

I realize this is kind of silly and opaque, but it's a low-effort way to get the right behavior by default…

Reply:
oh hmm...i added this because scripts are loaded in a random order since I'm using a hashmap. Without a hashmap, I suppose the order would be based on whatever order Rust enumerates file names in a dir. Maybe we need some way to define an order. or maybe we make it an error if names clash?

Reply:
Ah, right, that does actually make sense. What do you think about sorting the files by filename here?
https://github.com/calyxir/calyx/pull/2084/files#diff-174ce06968637c543eab68bb08eb4b7d31ce79432ae54293a1f3d0874a53d962R208-R217

Then, it seems like at least the declarations of things will happen in a deterministic order… and if we are lucky, they happen in, like, `import`-dependent order, which makes things seem sensible?